en: add OpenAPI docs (#822)
pingcap/docs-dm: en/open-api.md

---
title: Operate the DM Cluster Using OpenAPI
summary: Learn about how to use OpenAPI interface to manage the cluster status and data replication.
---

# Operate the DM Cluster Using OpenAPI

> **Warning:**
>
> DM OpenAPI is still an experimental feature and disabled by default. It is not recommended to use it in a production environment.

DM provides the OpenAPI feature for querying and operating the DM cluster, which is similar to the feature of [dmctl tools](./dmctl-introduction.md). If you need to enable this feature, add the following configuration in the DM-master configuration file:

```toml

> [experimental-features]

Reviewer: ```suggestion
[experimental]
```
sorry aobot this change, after internal discussions, we choose use `experimental` instead of `experimental-features`, and i forget update the pr...

please also update this part in docs-cn 😘 😘 😘

Author: ditto~

openapi = true
```

> **Note:**
>

> > - DM provides the [specification document](https://github.com/pingcap/ticdc/blob/master/dm/openapi/spec/dm.yaml) that meets OpenAPI 3.0.0 standard. This document contains all the request parameters and returned values and you can copy the document and preview it in [Swagger Editor](https://editor.swagger.io/).

Reviewer: ```suggestion
> - DM provides the [specification document](https://github.com/pingcap/ticdc/blob/master/dm/openapi/spec/dm.yaml) that meets OpenAPI 3.0.0 standard. This document contains all the request parameters and returned values and you can copy the document yaml and preview it in [Swagger Editor](https://editor.swagger.io/).
```

>

> > - The listening port of the DM-master node is 8216 by default. After you deploy the DM-master nodes, you can access `http://{host}:{port}/api/v1/docs` to preview the documentation online.

Reviewer: yes... 8261 is not default port.. it;s just the default port in our test-case😂😂😂, i will add some suggestion to fix this

> > - The listening port of the DM-master node is 8216 by default. After you deploy the DM-master nodes, you can access `http://{host}:{port}/api/v1/docs` to preview the documentation online.

Reviewer: ```suggestion
> - After you deploy the DM-master nodes, you can access `http://{master-addr}/api/v1/docs` to preview the documentation online.
```

it's my bad, please apply this change to docs-zh 😬😬😬😬

Author: fixed in https://github.com/pingcap/docs-dm/pull/822/commits/aa04eeaf2b89de9939a401342a46783385ab405b. PTAL~

You can use the APIs to perform the following maintenance operations on the DM cluster:

## APIs for managing clusters

> * [Get the status information of a DM-master node](#get-the-status-information-of-a-dm-master-node)

Reviewer: ```suggestion
* [Get the node information of a DM-master node](#get-the-status-information-of-a-dm-master-node)
```

it's some node infomation of dm-master, such as ip addr, peer port, so i think use status is not accuracy

Reviewer: please also change the dm-worker part

Author: how about just delete the `status`? Because there is already a `node` in the end.

Reviewer: lgtm

Author: fixed in https://github.com/pingcap/docs-dm/pull/822/commits/d436951ad44f24f29550c62f70e7b67e77f6373a.

* [Stop a DM-master node](#stop-a-dm-master-node)
* [Get the status information of a DM-worker node](#get-the-status-information-of-a-dm-worker-node)
* [Stop a DM-worker node](#stop-a-dm-worker-node)

## APIs for managing data sources

* [Create a data source](#create-a-data-source)
* [Get the data source list](#get-the-data-source-list)
* [Delete the data source](#delete-the-data-source)
* [Get the status information of a data source](#get-the-status-information-of-a-data-source)

> * [Start the relay-log feature for a data source](#start-the-relay-log-feature-for-a-data-source)

Reviewer: ```suggestion
* [Start the relay-log feature for data sources](#start-the-relay-log-feature-for-a-data-source)
```
in api we can operate relay log feature for many sources not only a source, stop/pause/resume is also the same way

Author: fixed in https://github.com/pingcap/docs-dm/pull/822/commits/a01e78f829350679ef0ee06cd94365a510db14a5. PTAL~
